backend/api/views/computetask.py

```diff
@@ -276,21 +287,24 @@ def get_queryset(self):
         )
 
 
-class ComputeTaskViewSet(ComputeTaskViewSetConfig, mixins.RetrieveModelMixin, mixins.ListModelMixin, GenericViewSet):
-    pass
-
-
 class CPTaskViewSet(ComputeTaskViewSetConfig, mixins.ListModelMixin, GenericViewSet):
+    serializer_class = ComputeTaskSerializer
+
     def get_queryset(self):
         compute_plan_key = self.kwargs.get("compute_plan_pk")
         validate_key(compute_plan_key)
 
-        queryset = super().get_queryset()
-        return queryset.filter(compute_plan__key=compute_plan_key).annotate(
-            # Using 0 as default value instead of None for ordering purpose, as default
-            # Postgres behavior considers null as greater than any other value.
-            duration=models.Case(
-                models.When(start_date__isnull=True, then=0),
-                default=Extract(Coalesce("end_date", Now()) - models.F("start_date"), "epoch"),
+        return (
+            ComputeTask.objects.filter(channel=get_channel_name(self.request))
```

Comment:
Is there a difference between using `ComputeTask.objects` and `super().get_queryset()`?

Reply:
I don't think so, I feel like it's slightly more explicit not to use `super()` given that there is some multiple inheritance with mixins here, but maybe I'm missing some under-the-hood optimization...